### PR TITLE
Added hoogle plugin

### PR DESCRIPTION
**Pull request by wz1000**

### CODE REVIEW

**Comment by alanz** [comment]
Can you modify this to use the library approach instead, now that [Hoogle 5.0.11](https://hackage.haskell.org/package/hoogle-5.0.11/docs/Hoogle.html) is out which provides `defaultDatabaseLocation`?

You will probably have to add it as an extra dependency at the end of  `stack.yaml`

File: hie-hoogle/Haskell/Ide/HooglePlugin.hs [hunk 1]
```diff
@@ -0,0 +1,91 @@
...
+        db <- defaultDatabaseLocation
+        dbExists <- doesFileExist db
+        if dbExists then do
+            res <- searchHoogle db expr
+            if null res then
+                return $ IdeResponseOk "No results found"
+            else do
+                let Target{..} = head res
+                    packageModule = unwords $ map fst $ catMaybes [targetPackage, targetModule]
+                return $ IdeResponseOk $ T.pack $ unlines $ [unHTML targetItem, packageModule, unHTML targetDocs]
```
Comment (alanz) [inline hie-hoogle/Haskell/Ide/HooglePlugin.hs]: Is this different from `disp`? Perhaps use `disp` here instead.

File: hie-hoogle/Haskell/Ide/HooglePlugin.hs [hunk 2]
```diff
@@ -0,0 +1,91 @@
...
+  }
+
+-- ---------------------------------------------------------------------
+
+infoCmd :: CommandFunc T.Text
+infoCmd = CmdSync $ \_ctxs req -> do
+  case getParams (IdText "expr" :& RNil) req of
+    Left err -> return err
+    Right (ParamText expr :& RNil) -> liftIO $ do
+        db <- defaultDatabaseLocation
```
Comment (alanz) [inline hie-hoogle/Haskell/Ide/HooglePlugin.hs]: Both commands do the same process until they get their results.

Maybe put that process in a helper function?

File: hie-hoogle/Haskell/Ide/HooglePlugin.hs (hunk 1 above)
Comment (wz1000) [inline hie-hoogle/Haskell/Ide/HooglePlugin.hs]: It is different from `disp`.

**Pull request merged**